homeassistant/components/sensor/google_travel_time.py

```diff
@@ -23,47 +25,88 @@
 CONF_ORIGIN = 'origin'
 CONF_DESTINATION = 'destination'
 CONF_TRAVEL_MODE = 'travel_mode'
+CONF_OPTIONS = 'options'
+CONF_MODE = 'mode'
+CONF_NAME = 'name'
 
 PLATFORM_SCHEMA = vol.Schema({
     vol.Required('platform'): 'google_travel_time',
+    vol.Optional(CONF_NAME): vol.Coerce(str),
     vol.Required(CONF_API_KEY): vol.Coerce(str),
     vol.Required(CONF_ORIGIN): vol.Coerce(str),
     vol.Required(CONF_DESTINATION): vol.Coerce(str),
-    vol.Optional(CONF_TRAVEL_MODE, default='driving'):
-        vol.In(["driving", "walking", "bicycling", "transit"])
+    vol.Optional(CONF_TRAVEL_MODE):
+        vol.In(["driving", "walking", "bicycling", "transit"]),
+    vol.Optional(CONF_OPTIONS): vol.All(
+        dict, vol.Schema({
+            CONF_MODE: cv.string,
+            'language': cv.string,
+            'avoid': cv.string,
+            'units': cv.string,
+            'arrival_time': cv.string,
+            'departure_time': cv.string,
+            'traffic_model': cv.string,
+            'transit_mode': cv.string,
+            'transit_routing_preference': cv.string
+        }))
 })
 
```

Comment:
For most of the config parameters, only a few values are valid, so the config validator should only accept valid values